plugin/src/main/java/com/google/errorprone/xplat/checker/JodaTimeLocal.java

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.errorprone.xplat.checker;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.NewClassTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.List;
+
+/**
+ * Bans the use of some Joda-Time constructors and methods that convert to a DateTime or use a
+ * DateTimeZone. A fix is suggested that is cross platform compatible.
+ */
+@BugPattern(
+    name = "JodaTimeLocal",
+    summary = "Bans the usage of timezoned toDateTime methods in some Joda-Time classes.",
+    explanation =
+        "The usage of timezoned LocalDateTime, LocalDate and LocalTime Joda-Time constructors"
+            + " and toDateTime methods are banned from cross platform development"
+            + " due to incompatibilities. A fix using a new DateTime is suggested.",
+    severity = ERROR)
+public class JodaTimeLocal extends BugChecker implements MethodInvocationTreeMatcher,
+    NewClassTreeMatcher {
+
+  private static final ImmutableMap<String, ImmutableList<String>> CLASS_NAMES =
```

Comment:
Readability comment: `CLASS_NAMES` doesn't capture the intent here. How about something along the lines of "DISALLOWED_CLASS_METHOD_MAP"?

Reply:
Sounds good